### PR TITLE
Add vagrant machine for auto tests

### PR DESCRIPTION
**Pull request by vogdb**

### CODE REVIEW

**Comment by vogdb** [comment]
@tomaszdurka please review

**Comment by vogdb** [comment]
@ppp0 please review

File: Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
+Vagrant.configure('2') do |config|
+  config.ssh.forward_agent = true
+  config.vm.box = 'cargomedia/debian-7-amd64-default'
+
+  config.vm.hostname = 'janus-gateway-js.dev.cargomedia.ch'
+
+  config.vm.network :private_network, ip: '10.10.10.16'
+  config.vm.synced_folder '.', '/home/vagrant/janus-gateway-js', :type => 'nfs'
```
Comment (ppp0) [inline Vagrantfile]: No need for `:type => 'nfs'` here, it actually complicates things (asks for password when vagrantupping)

File: Vagrantfile [hunk 2]
```diff
@@ -0,0 +1,20 @@
+Vagrant.configure('2') do |config|
+  config.ssh.forward_agent = true
+  config.vm.box = 'cargomedia/debian-7-amd64-default'
+
+  config.vm.hostname = 'janus-gateway-js.dev.cargomedia.ch'
+
+  config.vm.network :private_network, ip: '10.10.10.16'
```
Comment (ppp0) [inline Vagrantfile]: Since you won't be using any connection to the box besides `vagrant ssh` you can skip line 7 entirely

**Comment by ppp0** [comment]
I would say it LGTM

I'm not sure, tough, if you shouldn't add some test here as well to verify the proper test setup?

**Comment by vogdb** [comment]
@ppp0 please review

**Comment by ppp0** [comment]
LGTM, as discussed, we only add a Vagrant Box here ( @vogdb doesn't want to add tests at this point)

**Comment by ppp0** [comment]
Also be aware that in a not too far future, we will use Debian Jessie for production Janus

File: puppet/environments/development/manifests/site.pp [hunk 3]
```diff
@@ -0,0 +1,13 @@
+node default {
+
+  class { ['nodejs', 'chromium']: }
+
+  janus::role::standalone { 'janus':
+    hostname => 'janus-gateway-js.dev.cargomedia.ch',
+  }
+
+  package { ['gulp', 'testem']:
```
Comment (tomaszdurka) [inline puppet/environments/development/manifests/site.pp]: I would suggest to use local gulps, grunts, etc :)

Comment (vogdb) [inline puppet/environments/development/manifests/site.pp]: What do you mean by that? Sould I change the provider or what?

Comment (tomaszdurka) [inline puppet/environments/development/manifests/site.pp]: Oh I just meant that here we install `gulp` globally, with latest (or whatever defined in puppet package) version. I assume it's safer to install nodejs deps for project using `package.json` and run them locally - `node_modules/.bin/gulp`. This way you sure your gulpfile is in the same version as gulp binary.

Same goes for `browserify` in `package.json`. We use global binary (?), while we could/should use local one.

Please correct me if I'm wrong.

**Pull request merged**